cli/src/funTest/kotlin/ExamplesFunTest.kt

```diff
@@ -42,7 +42,7 @@ import org.ossreviewtoolkit.model.config.RepositoryConfiguration
 import org.ossreviewtoolkit.model.config.Resolutions
 import org.ossreviewtoolkit.model.licenses.LicenseConfiguration
 import org.ossreviewtoolkit.model.readValue
-import org.ossreviewtoolkit.model.utils.SimplePackageConfigurationProvider
+import org.ossreviewtoolkit.model.utils.createLicenseInfoResolver
```

Comment:
commit:  Not calling the rules per license source has the following effects

- The code which previously went into the require block as precondition for the respective rule now goes into a for loop inside the rule. I believe one underlying idea of the rules API was to get rid of such loops.

- The matcher functions like `isExcluded()` become harder to re-use, and if I'm not mistaken become unusable in some particular scenarios. E.g. if you want to know whether the detected license is excluded.

Can you explain the rationale for this?

Reply:
>     * The code which previously went into the require block as precondition for the respective rule now goes into a for loop inside the rule. I believe one underlying idea of the rules API was to get rid of such loops.

This is not true, license rules still use the require block. You refer to changes in our internal rules that were done for other reasons I will explain in the internal review.

>     * The matcher functions like `isExcluded()` become harder to re-use, and if I'm not mistaken become unusable in some particular scenarios. E.g. if you want to know whether the detected license is excluded.

I don't understand, you can still check if a detected license is excluded.

> Can you explain the rationale for this?

The current API forces the user to create one violation per source if a problematic license is found in multiple sources, because you cannot check there is already a violation for the same license with another source. If the source does not matter, this should not be required. You also cannot implement rules that check multiple sources at once, for example if you want to create a violation if a license was declared but not detected. In general, I think that if the a license is handled differently depending on the source there should be separate rules for that, otherwise it is sufficient if the rule is called once per license.

However, I have reverted that change because it was too much work to convert our internal rules to that approach. I'd still like to make that change sometime in the future.

Reply:
> This is not true, license rules still use the require block. You refer to changes in our internal rules that were done for other reasons I will explain in the internal review.

I though that the effects the change has on our rules would be an effect to have in general on rule implementation, not just on our internal ones. So it seems I missed something.